src/main/java/ch/njol/skript/expressions/ExprPlugins.java

```diff
@@ -0,0 +1,81 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import org.bukkit.plugin.Plugin;
+
+import java.util.ArrayList;
+
+
+@Name("Plugins")
+@Description("Returns a list of the server's loaded plugins.")
+@Examples({"if the plugins list contains \"Vault\":",
+		"send \"Plugins (%size of loaded plugins%): %plugins%\" to player"})
+@Since("INSERT VERSION")
+public class ExprPlugins extends SimpleExpression<String> {
+	
+	static {
+		Skript.registerExpression(ExprPlugins.class, String.class, ExpressionType.SIMPLE, "[(all [[of] the]|the)] [loaded] plugins");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected String[] get(Event e) {
+		ArrayList<String> plugins = new ArrayList<>();
+		for (Plugin p : Bukkit.getPluginManager().getPlugins()) {
+			plugins.add(p.getName());
+		}
+		return plugins.toArray(new String[0]);
```

Comment:
The JVM will just re-initalize the String array since 0 is not a valid size for any amount of plugins.

A suggestion I have is just making the whole thing return from a stream
```java
return Bukkit.getPluginManager().getPlugins().stream().map(plugin -> plugin.getName()).toArray(String[]::new);
```